.github/workflows/task-build-madara.yml

```diff
@@ -104,6 +144,21 @@ jobs:
           echo "$HASH" > cairo_artifacts_hash.txt
           echo "Hash of Cairo artifacts is: $HASH (with timestamp: $TIMESTAMP)"
 
+      - name: Generate Bootstrapper artifacts hash
+        id: generate_bootstrapper_artifacts_hash
+        run: |
+          # Get current Unix timestamp for uniqueness
+          TIMESTAMP=$(date +%s)
+          # Generate hash of all files in cairo/target directory
+          FILES_HASH=$(find artifacts -type f -exec sha256sum {} \; | sort | sha256sum | awk '{ print $1 }')
+          # Combine timestamp with files hash for unique identifier
+          HASH=$(echo "$TIMESTAMP $FILES_HASH" | sha256sum | awk '{ print $1 }')
+          # Set output for use in subsequent steps
+          echo "hash=$HASH" >> $GITHUB_OUTPUT
```

Comment:
not sure if this env will collide with the other one from the previous step with the same name